### DISCUSSION TITLE
Is it using Babel? If so, can we customise the config?

### DISCUSSION THREAD

**Discussion by tavoyne** [remix-run/remix#2278] [Q&A] [answered]
Is our source code transpiled and polyfilled by Remix before being sent to the browser? Like is it using Babel, `@babel-preset-env` and `core-js` in the background?

If so, what the target browsers are? And can the Babel config be customised?

**Comment by sergiodxa** [comment, marked as answer]
Remix uses esbuild and not Babel. It's not possible to customize the esbuild config.

For browser support, you can see this docs page https://remix.run/docs/en/v1/guides/browser-support

**Comment by tavoyne** [comment]
Ok thanks. Any idea of what the esbuild target is?

**Reply by sergiodxa** [reply]
It's targeting ES2015

**Comment by mattste** [comment]
Is there a workaround for this? My team is evaluating Remix but the lack of babel/compiler hooks support seems to be a barrier. We'd like to colocate our GraphQL queries in our code using something like Relay. It [requires a babel plugin](https://relay.dev/docs/getting-started/installation-and-setup/#set-up-babel-plugin-relay) to do so. 

I'm not opposed to even having to write an esbuild plugin for it but the complete lack of compiler hooks is very limiting. I understand the hesitation with committing to a public API. Hopefully tools such as Vite can provide some inspiration.

**Reply by sergiodxa** [reply]
The compiler API is still not public and it will most likely not be for a long time, they want to be able to switch the compiler without doing a breaking change (they already did it once).

Relay doesn't work directly with Remix sadly, but there's a hacky way, create a `src` folder along your `app` folder, install Babel and the Relay plugin for it and build your code from `src` to `app` and then let Remix build the `app` code (your Babel output), it will add a double build but that will allow you to use tools that require a Babel plugin.

Ideally, in a future, Relay may not need a Babel plugin and it will let you compile your queries directly with another tool in parallel to your Remix app like Tailwind CLI let's you do that for the CSS.

**Comment by MiguelNiblock** [comment]
I'm exploring remix for the prospect of building universal apps with react-native-web. And the lack of compiler config is a barrier to use some common react-native libraries that work on web, with babel. For example, UI component libraries like react-native-elements, native-base or react-native-paper. These require babel config to work in web. So I guess they won't work with remix?

**Reply by kiliman** [reply]
You can build your react native code using babel separate from Remix. As long as it can generate a cjs or esm bundle you can import into your remix app. The beauty of Remix is that it doesn’t try to handle everything but you can orchestrate many different tools to build your app.

**Reply by MiguelNiblock** [reply]
That's an interesting approach. Thank you again @kiliman for supporting us in these discussions. As I said here https://github.com/remix-run/remix/discussions/1578#discussioncomment-2839216, I would love to put this approach into practice but I'm not very familiar with it, although it sounds simple enough, since technically I should be able to just reuse the same webpack/babel configuration I already have for transpiling on-the-fly, correct? The only difference is it would need to be done statically, and only once, which seems like it's a much more efficient approach than transpiling every time there's changes, especially since these 3rd party dependencies won't be changing every time because they're not part of my application code.

**Reply by kiliman** [reply]
If you can create a simple react native web app with navigation, I can try and help you with the integration work.

**Comment by MiguelNiblock** [comment]
To put that pre-compile approach into practice, perhaps I could create a git subproject, so it is part of my main git repo, but it can be updated independently. Then create a dist/ folder within it for the compiled code.

**Comment by luispuig** [comment]
Did anyone test the "compiling with babel" approach? Any repo to check how it was done? We are running similar issues.

**Comment by MiguelNiblock** [comment]
@luispuig I did not pursue the approach, because I've become busier recently. There's no examples available. Only comments from remix guys saying it should be possible in theory, but I think it's not their priority to support configuring the web build, from their design perspective. I lost interest in remix for this reason.

**Reply by luispuig** [reply]
Thanks for the answer, I'll dig in a bit more.
